avro-serializer/src/main/java/io/confluent/kafka/serializers/KafkaAvroDeserializer.java

```diff
@@ -53,6 +54,11 @@ public Object deserialize(String s, byte[] bytes) {
     return deserialize(bytes);
   }
 
+  /**
+   * Pass a reader schema to get an Avro projection
+   */
+  public Object deserialize(String s, byte[] bytes, Schema readerSchema) { return deserialize(bytes, readerSchema); }
```

Comment:
Not sure I understand how this is supposed to be used in a normal application since this is a new variant of `deserialize` that the consumer will never invoke. Same for the Decoder implementation. This would allow you to use the deserializer the way you describe if you do it all manually, but that doesn't seem all that useful -- most people don't drive the deserializer manually.

One way to do this would be to have a setting similar to the one used to specify the SpecificRecord type and get the Schema from that.